commands/addrole.js

```diff
@@ -0,0 +1,127 @@
+        ).then(r => r.delete(30000))
+    }
+    // check if there's an emoji already bound to that message with a role.
+    reaction_message_model.findOne({
+        messageID: message_id
+    }, async (err, reaction_message) => {
+        msg = await message.reply("Loading...")
+        if (err) console.log(err)
+        // create a new object of the model to add it to the database
+        if (!reaction_message) {
+            const new_reaction_message = new reaction_message_model({
+                guildID: message.guild.id,
+                channelID: channel_id,
+                messageID: message_id,
+                roles: {}
+            })
+            new_reaction_message.roles[emoji] = {}
+            new_reaction_message.roles[emoji]["name"] = role.name
+            new_reaction_message.roles[emoji]["id"] = role.id
+            new_reaction_message.save().catch(err => console.log(err))
+        } else {
+            // check if the same emote we're trying to add already exists in the database
+            if (emoji in reaction_message.roles) {
+                const filter = m => m.author.id === message.author.id
+                // ask the user if they want to overwrite the current role that's bounded to this emoji in the database
+                let overwrite = new Discord.RichEmbed()
+                    .setColor("#FFA500")
+                    .addField("Warning", `The emote ${emoji} was already found in the database on the mentioned message for the role **${role.name}**, Would you like to overwrite it?`)
+                    .addField("Respone", "Type **Yes** to overwrite or anything else to Cancel")
+                    .setDescription("*This command will be automatically cancelled in 60 seconds*")
+                msg.edit(overwrite)
+                // await for the user's reply for 60 seconds.
+                var eternity = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
+                    if (!collected.first()) {
+                        msg.edit(clean.setColor("#FF0000").addField("Failture", "Command Timed Out")).then(m => m.delete(30000))
+                        return false
+                    } else {
+                        if (collected.first().content.toLowerCase() == "yes") {
+                            msg.edit(new Discord.RichEmbed().setColor("#F0FFFA").addField("Information", "Emote will be overwritten, please wait"))
+                            return true
+                        } else {
+                            msg.edit(clean.setColor("#FFFF00").addField("Information", "Command Cancelled.")).then(m => m.delete(30000))
+                            return false
+                        }
+                    }
+                }).catch(r => { msg.edit(clean.setColor("#FFFF00").addField("Information", "Command Cancelled.")).then(m => m.delete(30000)); console.error; return false })
```

Comment:
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.

Post-reformat, it's easier to see that there's something incorrect - `console.error` is mentioned with no function call, or arguments. Perhaps call it with `r` as an arg?
```suggestion
			}).catch(function(r) {
				msg.edit(clean.setColor("#FFFF00")
					.addField("Information", "Command Cancelled."))
					.then(m => m.delete(30000));
				console.error(r);
				return false;
			});
```